instance/tests/integration/integration_instance.py

```diff
@@ -489,12 +490,22 @@ def test_openstack_server_terminated(self):
         """
         Test that OpenStackServer detects if the VM was terminated externally.
         """
-        server = OpenStackServer(name_prefix="integration_test")
-        server.save()
-        server.start()
-        server.sleep_until(lambda: server.status.accepts_ssh_commands, timeout=120)
-        server.os_server.delete()
-        self.assert_server_terminated(server)
+        retries = 1
+        while retries >= 0:
+            server = OpenStackServer(name_prefix="integration_test")
+            server.save()
+            server.start()
+            try:
+                server.sleep_until(lambda: server.status.accepts_ssh_commands, timeout=120)
+            except TimeoutError:
+                server.os_server.delete()
+                retries -= 1
+            else:
+                server.os_server.delete()
+                self.assert_server_terminated(server)
+                break
+
+        self.assertTrue(retries >= 0)
```

Comment:
We don't need to assert on the number of retries since that's just a test implementation detail.